### PR TITLE
Using mycelo for deploying testnets

### PR DESCRIPTION
**Pull request by jcortejoso**
### Description

Introducing `mycelo` as an option for deploying a testnet. If the new env variable `GETH_USE_MYCELO` equals to `true`, mycelo will be used to generate the genesis file for the network.
Currently a limited number of configuration can be used to configure the network, like the mnemonic, number of validators, fauceted accounts (`LOAD_TEST_CLIENTS`), blocktime, epoch period and network gas limit. A further PR celo-blockchain#1390 will introduce more  customization from the migrationConfig.js. It is also possible to modify the governable parameters once the network is deployed.

### Other changes

The genesis file sharing among nodes is done using a gcp bucket. Genesis file with migrations is far larger that Kubernetes configmap limit size (around 1MB). We could consider using a local file artifact as minio if we prefer to remove gcp dependency.

### Tested

Deployed testnet

### Related issues

- Fixes celo-blockchain#1383

### Backwards compatibility

No problem. Requires access to the gcp bucket (public bucket)

### Documentation

_The set of community facing docs that have been added/modified because of this change_

### CODE REVIEW

File: packages/celotool/src/lib/helm_deploy.ts [hunk 1]
```diff
@@ -1073,3 +1089,97 @@ function rollingUpdateHelmVariables() {
...
+  })
+  const genesisPath = path.join(celoBlockchainDir, 'genesis.json')
+  const genesisContent = fs.readFileSync(genesisPath).toString()
+  return genesisContent
+
+  // Clean up the tmp dir as it's no longer needed
+  await spawnCmd('rm', ['-rf', celoBlockchainDir], { silent: true })
+}
+
+export async function createGethConfigConfigMap(celoEnv: string, genesisJson: string) {
```
Comment (trianglesphere) [inline packages/celotool/src/lib/helm_deploy.ts]: Given that the genesis file is too large for a config map & that this isn't used anywhere, can this be removed or do you plan to use it later?

File: packages/helm-charts/common/templates/_helpers.tpl [hunk 2]
```diff
@@ -274,8 +274,10 @@ metadata:
   labels:
 {{ include "common.standard.labels" .  | indent 4 }}
 data:
-  networkid: {{ .Values.genesis.networkId | quote }}
-  genesis.json: {{ .Values.genesis.genesisFileBase64 | b64dec | quote }}
+  networkid: {{ $.Values.genesis.networkId | quote }}
+{{- if eq (default $.Values.genesis.useGenesisFileBase64 "true") "false" }}
```
Comment (trianglesphere) [inline packages/helm-charts/common/templates/_helpers.tpl]: IUUC this is saying that if `useGensisFileBase64 == false` then use the base 64 genesis file? That seems inverted from what it should be.

File: packages/celotool/src/lib/helm_deploy.ts [hunk 3]
```diff
@@ -1073,3 +1089,97 @@ function rollingUpdateHelmVariables() {
     )}`,
   ]
 }
+
+const celoBlockchainDir: string = path.join(os.tmpdir(), 'celo-blockchain-celotool')
+
+export async function getGenesis(celoEnv: string, reset: boolean, useMyCelo: boolean) {
```
Comment (trianglesphere) [inline packages/celotool/src/lib/helm_deploy.ts]: nit: the seems like it's `createAndUploadGenesisToGCP` (name's a little long), but it's odd that `getGenesis` doesn't return anything but instead place the genesis in google cloud.

File: packages/celotool/src/lib/helm_deploy.ts [hunk 4]
```diff
@@ -758,7 +774,7 @@ async function helmParameters(celoEnv: string, useExistingGenesis: boolean) {
     `--set celotool.image.repository=${fetchEnv('CELOTOOL_DOCKER_IMAGE_REPOSITORY')}`,
     `--set celotool.image.tag=${fetchEnv('CELOTOOL_DOCKER_IMAGE_TAG')}`,
     `--set domain.name=${fetchEnv('CLUSTER_DOMAIN_NAME')}`,
-    `--set genesis.genesisFileBase64=${Buffer.from(genesisContent).toString('base64')}`,
+    `--set genesis.useGenesisFileBase64="false"`,
```
Comment (trianglesphere) [inline packages/celotool/src/lib/helm_deploy.ts:777]: Are there any cases where this is true?

**Review by trianglesphere — COMMENTED** [review]
I left a couple small comments, but this is looking pretty good.

File: packages/celotool/src/lib/helm_deploy.ts (hunk 4 above)
Comment (jcortejoso) [reply packages/celotool/src/lib/helm_deploy.ts:777]: Not deploying with celotool, but I preferred to keep the compatibility on the helm chart if someone wants to use directly with helm (and do not introduce a hard dependency with google storage).

**Review by trianglesphere — APPROVED** [review]
looks good to me.

File: packages/celotool/src/lib/helm_deploy.ts (hunk 1 above)
Comment (jcortejoso) [reply packages/celotool/src/lib/helm_deploy.ts]: I do not discard exploring options to store the genesis alternatively to Google Storage in the short term, but for now I think we can remove this. Thanks for the comment.

**Pull request merged**